### ISSUE TITLE
Supports boards other than teensy 2.0

### ISSUE THREAD

**Issue by miraclehoshi** [pepijndevos/TCPoke#1]
I really like this project! But there is a problem. This must require teensy 2.0.

teensy 2.0 bootloader is not an open source, which means that compatible boards are not available to make it.

I don't know much about software development.
But can you add a code for this project to use an aduino micro or leonardo?

In my estimation, using NicoHood/HID is possible
(Maybe not. That's my guess.)

I'm sorry I can't speak English well. Please understand.
I really love this project! But teensy 2.0 is very expensive. There are high prices and overseas shipping costs.

Please consider it for me and everyone.

**Comment by pepijndevos** [comment]
I don't currently have any plans to update this software.
Since Chrome apps are deprecated, all that needs tot be rewritten.
In the next version could easily work over serial rather than HID.

**Comment by miraclehoshi** [comment]
Wow! Thank you for your answer! I look forward to it very much!

**Comment by miraclehoshi** [comment]
https://i.postimg.cc/yxXcQvSP/tcpokev1.png

I designed this. I'm waiting for the next version.

**Comment by pepijndevos** [comment]
That looks awesome! It looks way to accurate for a mockup, so is that an actual working design?
One piece of advice: The Game Link port is practically impossible to source, but a PCB of the right size just fits right into any random Game Link cable from eBay. So it's save a lot of trouble to just make the PCB shaped like the centre part of the Game Link port. I've seen some boards that do it.

Updating the software is not very high on my list of priorities, but it's all open source, so I'd totally accept a PR that makes the code work with your board.

**Comment by miraclehoshi** [comment]
it's actually a working circuit.
Combine the circuit in teensy 2.0 with the  TCPoke shield circuit.

I disassembled my AGB and reverse-engineered link connecter's footprint.
also bought 100 link cable connectors.
https://i.postimg.cc/66Z7XrmC/HTB1c9a-Dk-Qy-WBu-Njy0-Fpq6yss-XXae.jpg

so, I started making samples! I'll let you know when it's done.

**Comment by pepijndevos** [comment]
That's amazing!

Let me give you a few pointers on how to make the software work:

The Arduino code is actually pretty trivial, and could easily be replaced by another HID stack or just plain serial. More on that later.

https://github.com/pepijndevos/TCPoke/blob/6784c2954a98eb39871063e064f54277e2f09f22/tcpoke_teensy/tcpoke_teensy.ino#L34-L35
Pretty much this has to be changed to something like
```
if(Serial.available()) {
    data = Serial.read()
```
and similar to the send line a few lines further down.
Just make sure to `flush()`, or you might have a bad time.

Actually the bigger challenge is updating the desktop app.
It is a Chrome app written in React, and it uses the `chrome.hid` API.
https://github.com/pepijndevos/TCPoke/blob/6784c2954a98eb39871063e064f54277e2f09f22/tcpoke_chrome/jsx/control-panel.js#L331

Chrome apps are no longer supported. One solution would be to use NW.js, which supports the Chrome platform API. Last time I checked this it did not support the serial and HID API, but this appears to be supported now. So if a new HID stack is used, the client can just be rebuilt using NW.js, and if serial is used, it'd again be pretty doable to replace these calls with `chrome.serial`.

But Electron is what all the cool kids use these days, so I think eventually it's preferable to use that. Luckily React is not obsolete yet (you never know in the world of JS), so most of the app should be fine as it is.

This will get stuff to work as it was when I left it, with a new board and updated software. The final thing that would really seal the deal is improving the actual game support. Right now, gen 1 trading works very well, but battle desyncs sometimes due to RNG, which I try to patch, but apparently incorrectly:
https://github.com/pepijndevos/TCPoke/blob/6784c2954a98eb39871063e064f54277e2f09f22/tcpoke_chrome/jsx/control-panel.js#L418
I also attempted support for gen 2, but this is not yet functional. The process is very similar, but the initialisation is slightly different.
https://github.com/pepijndevos/TCPoke/blob/6784c2954a98eb39871063e064f54277e2f09f22/tcpoke_chrome/jsx/control-panel.js#L411

I currently have some other project going, but your work looks very promising. Let me know if I can do anything else to help you.
